plugin/src/main/groovy/com/novoda/staticanalysis/internal/ktlint/KtlintConfigurator.groovy

```diff
@@ -0,0 +1,97 @@
+    static KtlintConfigurator create(Project project,
+                                     NamedDomainObjectContainer<Violations> violationsContainer,
+                                     Task evaluateViolations) {
+        Violations violations = violationsContainer.maybeCreate('ktlint')
+        return new KtlintConfigurator(project, violations, evaluateViolations)
+    }
+
+    KtlintConfigurator(Project project, Violations violations, Task evaluateViolations) {
+        this.project = project
+        this.violations = violations
+        this.evaluateViolations = evaluateViolations
+        this.variantFilter = new VariantFilter(project)
+    }
+
+    @Override
+    void execute() {
+        project.extensions.findByType(StaticAnalysisExtension).ext.ktlint = { Closure config ->
+            if (!project.plugins.hasPlugin(KTLINT_PLUGIN)) {
+                throw new GradleException(KTLINT_NOT_APPLIED)
+            }
+
+            def ktlint = project.ktlint
+            ktlint.ignoreFailures = true
+            ktlint.reporters = ['CHECKSTYLE', 'PLAIN']
+            ktlint.ext.includeVariants = { Closure<Boolean> filter ->
+                variantFilter.includeVariantsFilter = filter
+            }
+            config.delegate = ktlint
+            config()
+
+            project.afterEvaluate {
+
+                project.plugins.withId("kotlin") {
+                    configureKotlinProject()
+                }
+                project.plugins.withId("kotlin2js") {
+                    configureKotlinProject()
+                }
+                project.plugins.withId("kotlin-platform-common") {
+                    configureKotlinProject()
+                }
+                project.plugins.withId('com.android.application') {
+                    configureAndroidWithVariants(variantFilter.filteredApplicationVariants)
+                }
+                project.plugins.withId('com.android.library') {
+                    configureAndroidWithVariants(variantFilter.filteredLibraryVariants)
+                }
+            }
+        }
+    }
+
+    private void configureKotlinProject() {
+        project.sourceSets.each { configureKtlint(it) }
+    }
+
+    private void configureAndroidWithVariants(def mainVariants) {
+        mainVariants.all { configureKtlint(it) }
+        variantFilter.filteredTestVariants.all { configureKtlint(it) }
+        variantFilter.filteredUnitTestVariants.all { configureKtlint(it) }
+    }
+
+    private void configureKtlint(def sourceSet) {
+        def collectViolations = createCollectViolationsTask(violations, sourceSet.name)
+        evaluateViolations.dependsOn collectViolations
+    }
+
+    private def createCollectViolationsTask(Violations violations, def sourceSetName) {
+        project.tasks.create("collectKtlint${sourceSetName.capitalize()}Violations", CollectCheckstyleViolationsTask) { task ->
+            task.xmlReportFile = new File(project.buildDir, "reports/ktlint/ktlint-${sourceSetName}.xml")
```

Comment:
No way to change the output in the plugin. Hardcoded for now.

Reply:
Does this mean we get `ktlint-main.xml`? Never used the underlying Ktlint plugin, so not sure :)

Reply:
I think Gradle plugin provides the file name and path internally to ktlint tool.

It has report per sourceSet in pure Kotlin, and report per variant in Android projects. 

The path and file names are hardcoded and not configurable by the user.